numba/tests/test_parfors.py

```diff
@@ -1939,6 +1939,18 @@ def test_impl(A):
         self.prange_tester(test_impl, A, scheduler_type='signed',
                            check_fastmath=True, check_fastmath_result=True)
 
+    @skip_unsupported
+    def test_prange25(self):
```

Comment:
There's a test below called  `test_prange25` which overwrites the definition here, as a result this test doesn't run. Also, the function body is invalid as `x` isn't defined and if that's fixed then the test doesn't pass.

This is the modified test with which I've been checking the validity of this patch:
```python
    def test_prange25(self):
        def test_impl(x):
            n = len(x)
            buf = [np.zeros_like(x) for _ in range(n)]
            for i in range(n):
                buf[i] = x + i
            return buf
        A = np.ones((10,))
        self.prange_tester(test_impl, A, scheduler_type='unsigned',
                           check_fastmath=True, check_fastmath_result=True)

        cpfunc = self.compile_parallel(test_impl, (numba.typeof(A),))
        diagnostics = cpfunc.metadata['parfor_diagnostics']
        hoisted_allocs = diagnostics.hoisted_allocations()
        self.assertEqual(len(hoisted_allocs), 0)
```